### PR TITLE
Bugfix: Conversion between Label and Image with original scaling

### LINKED ISSUE #3389: Convert to Image/Convert to Labels do not preserve scaling
## 🐛 Bug

Right click on label/image layers > Convert to Image/Convert to Labels do not preserve scaling.

## To Reproduce

1. Create an image layer, set the layer scale to something other than default [1,1,1]
2. Right click on the image layer > Convert to Labels
3. The new labels layer ignores the scaling and reverts back to default [1,1,1]  

Same behavior also goes from labels layer to image layer.

## Expected behavior

Convert to Image/Convert to Labels should maintain the same scaling as the original layer. 

## Environment
napari: 0.4.12.dev11+g5a43dc59
Platform: macOS-11.5.2-x86_64-i386-64bit
System: MacOS 11.5.2
Python: 3.8.2 (default, Jun 8 2021, 11:59:35) [Clang 12.0.5 (clang-1205.0.22.11)]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.20.3
SciPy: 1.7.1
Dask: 2021.09.0
VisPy: 0.8.1

OpenGL:
- GL version: 2.1 ATI-4.6.20
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1792x1120, scale 2.0

Plugins:
- bioformats: 0.2.1
- cellpose-napari: 0.1.3
- console: 0.0.4
- napari-features: 0.1.4
- ome-types: 0.2.9
- scikit-image: 0.4.12.dev11+g5a43dc59
- svg: 0.1.5

### PR DESCRIPTION
**Pull request by 2dx**
# Description
Adding a change to fix https://github.com/napari/napari/issues/3389

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] loading image with `napari.view_image(skimage.data.moon(), scale=[2, 1])` and convert it to Label and back to Image.  

Could anyone tell me what test is required and how to run tests?

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by tlambert03 — APPROVED** [review]
welcome @2dx!  Great fix, thanks!

**Comment by tlambert03** [comment]
> Could anyone tell me what test is required and how to run tests?

You could update [this test here](https://github.com/napari/napari/blob/f0e83c27816c94944fc84ce130feb5c82af0fe2d/napari/layers/_tests/test_layer_actions.py#L100-L105) to change some attribute about the `input` layer and make sure it carries over to the new layer.

Tests will run here on CI, but to test it locally you can run `pytest`.  That will take a long time though, so to direct it only to the relevant test, run:

```
pytest napari/layers/_tests/test_layer_actions.py
```

(this assumes you've installed the testing dependencies with `pip install -e .[testing]`

**Comment by 2dx** [comment]
Thank you @tlambert03 
It is not related to this PR but I got error when running the test: pytest napari/layers/_tests/test_layer_actions.py
[log.txt](https://github.com/napari/napari/files/8262535/log.txt)

**Comment by tlambert03** [comment]
huh, haven't seen that error before.  Not quite sure what to tell you.  would be curious to know if you still got it in a fresh environment.  Was this a relatively new environment?

**Comment by 2dx** [comment]
I tried with absolutely fresh environment. Also I tried in anaconda environment and system python. If there is docker image that can run test I can use it.

**Comment by 2dx** [comment]
Finally I can run and add test with new WSL2 environment. I don't know what was wrong but thank you anyway.

File: napari/layers/_tests/test_layer_actions.py [hunk 1]
```diff
@@ -99,7 +99,10 @@ def test_convert_dtype(mode):
 )
 def test_convert_layer(input, type_):
     ll = LayerList()
+    input.scale *= 1.5
+    original_scale = input.scale.copy()
     ll.append(input)
     assert ll[0]._type_string != type_
     _convert(ll, type_)
     assert ll[0]._type_string == type_
+    assert (ll[0].scale == original_scale).all()
```
Comment (tlambert03) [inline napari/layers/_tests/test_layer_actions.py]:
```suggestion
    assert np.array_equal(ll[0].scale, original_scale)
```

**Review by tlambert03 — APPROVED** [review]
thanks @2dx!

**Comment by jni** [comment]
The error:

```pytb
..\..\..\anaconda3\envs\naparidev\lib\site-packages\pluggy\_manager.py:91: in register
    raise ValueError(
E   ValueError: Plugin already registered: c:\users\yukim\documents\napari-dev2\napari\napari\conftest.py=<module 'napari.conftest' from 'c:\\users\\yukim\\documents\\napari-dev2\\napari\\napari\\conftest.py'>
```

to me looks like an issue with pytest Python paths. Sometimes I have to do `PYTHONPATH=. pytest path/to/test.py`, never really understood why, but give it a go? How did you install napari for development? `pip install -e .[all,dev]`?

**Pull request merged**